### PR TITLE
Added pagination on vulnerability collection view

### PR DESCRIPTION
**Pull request by Cheshiriks**

### CODE REVIEW

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/vulnerability/VulnerabilityController.kt [hunk 1]
```diff
@@ -56,18 +56,42 @@ class VulnerabilityController(
...
     fun getAllVulnerabilities(
+        @RequestParam page: Int,
+        @RequestParam size: Int,
         @RequestBody filter: VulnerabilityFilter,
         authentication: Authentication?,
     ): Mono<VulnerabilityMetadataDtoList> = blockingToMono {
         if (
             // if user is not authenticated, he will have authentication = null and will not get other's submitted vulnerabilities
-            filter.status != VulnerabilityStatus.APPROVED && authentication?.name != filter.authorName &&
+            filter.statuses?.contains(VulnerabilityStatus.APPROVED) != true && authentication?.name != filter.authorName &&
```
Comment (akuleshov7) [inline save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/vulnerability/VulnerabilityController.kt:66]: what's about AUTO_APPROVED

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/vulnerability/VulnerabilityController.kt [hunk 2]
```diff
@@ -56,18 +56,42 @@ class VulnerabilityController(
...
+        description = "Get filtered vulnerabilities.",
+    )
+    @ApiResponse(responseCode = "200", description = "Successfully fetched all vulnerabilities by filters")
+    fun getCountVulnerabilities(
+        @RequestBody filter: VulnerabilityFilter,
+        authentication: Authentication?,
+    ): Mono<Long> = blockingToMono {
+        if (
+            // if user is not authenticated, he will have authentication = null and will not get other's submitted vulnerabilities
+            filter.statuses?.contains(VulnerabilityStatus.APPROVED) != true && authentication?.name != filter.authorName &&
```
Comment (akuleshov7) [inline save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/vulnerability/VulnerabilityController.kt:88]: AUTO_APPROVED again?

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/vulnerability/VulnerabilityController.kt [hunk 3]
```diff
@@ -56,18 +56,42 @@ class VulnerabilityController(
     )
     @ApiResponse(responseCode = "200", description = "Successfully fetched all vulnerabilities by filters")
     fun getAllVulnerabilities(
+        @RequestParam page: Int,
+        @RequestParam size: Int,
         @RequestBody filter: VulnerabilityFilter,
         authentication: Authentication?,
     ): Mono<VulnerabilityMetadataDtoList> = blockingToMono {
         if (
```
Comment (kgevorkyan) [inline save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/vulnerability/VulnerabilityController.kt:64]: may be extract this part into some method? its duplicated in 2 places

**Comment by kgevorkyan** [comment]
could you also add screenshot

File: save-cosv/src/main/kotlin/com/saveourtool/save/cosv/service/VulnerabilityMetadataService.kt [hunk 4]
```diff
@@ -57,9 +59,30 @@ class VulnerabilityMetadataService(
...
+            addRating(metadata)
+        }
+
         return vulnerabilityMetadataRepository.save(metadata)
     }
 
+    /**
+     * @param metadata
+     */
+    fun addRating(metadata: VulnerabilityMetadata) {
```
Comment (nulls) [inline save-cosv/src/main/kotlin/com/saveourtool/save/cosv/service/VulnerabilityMetadataService.kt:73]: do we need it as a public method? if yes, can we mark it as transactional ?

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/vulnerability/VulnerabilityController.kt (hunk 1 above)
Comment (Cheshiriks) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/vulnerability/VulnerabilityController.kt]: Fixed

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/vulnerability/VulnerabilityController.kt (hunk 2 above)
Comment (Cheshiriks) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/vulnerability/VulnerabilityController.kt]: Fixed

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/vulnerability/VulnerabilityController.kt (hunk 3 above)
Comment (Cheshiriks) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/vulnerability/VulnerabilityController.kt]: Fixed

**Comment by Cheshiriks** [comment]
> could you also add screenshot

Why a screenshot? There are no changes on the frontend.

**Review by nulls — APPROVED** [review]

**Comment by kgevorkyan** [comment]
> > could you also add screenshot
> 
> Why a screenshot? There are no changes on the frontend.

but you add pagination
anyway, nevermind

**Comment by Cheshiriks** [comment]
> > > could you also add screenshot
> > 
> > 
> > Why a screenshot? There are no changes on the frontend.
> 
> but you add pagination anyway, nevermind

There were pages on the front before. Pagination simply sends a request to the backend after each page is clicked.

**Pull request merged**